client/sites/components/sites-dashboard.tsx

```diff
@@ -183,30 +183,34 @@ const SitesDashboard = ( {
 					],
 			  }
 			: {} ),
-		type: selectedSite ? 'list' : 'table',
-		layout: {
-			styles: {
-				site: {
-					width: getSiteNameColWidth( isDesktop, isWide ),
-				},
-				plan: {
-					width: '126px',
-				},
-				status: {
-					width: '142px',
-				},
-				'last-publish': {
-					width: '146px',
-				},
-				stats: {
-					width: '106px',
-				},
-				actions: {
-					width: '74px',
-				},
-			},
-		},
-	} as View;
+		...( selectedSite
+			? { type: 'list', layout: {} }
+			: {
+					type: 'table',
+					layout: {
+						styles: {
+							site: {
+								width: getSiteNameColWidth( isDesktop, isWide ),
+							},
+							plan: {
+								width: '126px',
+							},
+							status: {
+								width: '142px',
+							},
+							'last-publish': {
+								width: '146px',
+							},
+							stats: {
+								width: '106px',
+							},
+							actions: {
+								width: '74px',
+							},
+						},
+					},
+			  } ),
```

Comment:
This seems like it might have been the original source of many of the problems. The `defaultDataViewsState` type never really conformed to the `View` type since it was possible to have a `list` type with `styles`, but that should never be possible.